src/main/java/appeng/util/ItemSorters.java

```diff
@@ -54,38 +52,30 @@ public int compare( final IAEItemStack o1, final IAEItemStack o2 )
 		{
 			final AEItemStack op1 = (AEItemStack) o1;
 			final AEItemStack op2 = (AEItemStack) o2;
+			int cmp = op1.getModID().compareToIgnoreCase( op2.getModID() );
 
-			if( getDirection() == SortDir.ASCENDING )
+			if( cmp == 0 )
 			{
-				return this.secondarySort( op2.getModID().compareToIgnoreCase( op1.getModID() ), o1, o2 );
+				cmp = Platform.getItemDisplayName( o1 ).compareToIgnoreCase( Platform.getItemDisplayName( o2 ));
 			}
-			return this.secondarySort( op1.getModID().compareToIgnoreCase( op2.getModID() ), o2, o1 );
-		}
 
-		private int secondarySort( final int compareToIgnoreCase, final IAEItemStack o1, final IAEItemStack o2 )
-		{
-			if( compareToIgnoreCase == 0 )
-			{
-				return Platform.getItemDisplayName( o2 ).compareToIgnoreCase( Platform.getItemDisplayName( o1 ) );
-			}
-
-			return compareToIgnoreCase;
+			return applyDirection( cmp );
 		}
 	};
+
 	public static final Comparator<IAEItemStack> CONFIG_BASED_SORT_BY_SIZE = new Comparator<IAEItemStack>()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1604)